app/src/main/java/ru/mikhailskiy/intensiv/network/MovieApiClient.kt

```
@@ -1,5 +1,39 @@
 package ru.mikhailskiy.intensiv.network
 
+import com.google.gson.GsonBuilder
+import io.reactivex.schedulers.Schedulers
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.gson.GsonConverterFactory
+import ru.mikhailskiy.intensiv.BuildConfig
+
+
 object MovieApiClient {
 
+    //todo add cache
+
+    private val client: OkHttpClient = OkHttpClient.Builder().apply {
+        if(BuildConfig.DEBUG) {
+            addInterceptor(HttpLoggingInterceptor().apply {
+                this.level = HttpLoggingInterceptor.Level.BODY
+            })
+        }
+    }.build()
+
+    val apiClient: MovieApiInterface by lazy{
+        val gson = GsonBuilder()
+            .setLenient()
```

С одной стороны прикольная фича, с другой стороны есть такой принцип Fail fast - то есть если что то работает не так - то лучше узнать об этом как можно раньше. C этим параметром - можно долго потом искать причину какой-нибудь ошибки. Так что использовать стоит с осторожностью